pkgs/jupiter-hw-support/default.nix

```
@@ -0,0 +1,115 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, autoPatchelfHook
+, makeWrapper
+, python3
+, xorg
+
+# jupiter-biosupdate
+, libkrb5
+, zlib
+, jq
+, dmidecode
+}:
+
+let
+  pythonEnv = python3.withPackages (py: with py; [
+    evdev
+    crcmod
+    click
+    progressbar
```

```suggestion
    progressbar2
```

¯\\\_(ツ)\_/¯

https://stackoverflow.com/a/57996842

* * *

```
$ sudo jupiter-controller-update
PTS: 1628633978, STS: 0
[
  {
    "path": "/dev/hidraw1",
    "vendor_id": 10462,
    "product_id": 4100,
    "serial_number": "N/A",
    "release_number": 256,
    "manufacturer_string": "Valve Software",
    "product_string": "Steam Controller Bootloader",
    "usage_page": 65280,
    "usage": 1,
    "interface_number": 0,
    "build_timestamp": 1628633978,
    "secondary_build_timestamp": 0,
    "is_bootloader": true
  }
]
Device is in bootloader mode, update required
Found candidate device, build timestamp 1628633978, BL true, BL_Type 1, HYB false
Updating Type 1 System
Erasing: |                                                                                    
Programming: [################################################################] 179.0 KiB 100%
Waiting for app to enumerate: |                                                               
SUCCESS
Firmware updated to 1658426874
```